File: codes/mathematica/p080.m
(* 
 * Solution to Project Euler problem 80
 * Copyright (c) Sankha Saha. All rights reserved.
 * 
 *  
 *   
 *)


DecimalPrefix[x_, n_] := Floor[x * 10^(n - 1 - Floor[Log[10, x]])]
DecimalPrefixDigitSum[x_, n_] := Total[IntegerDigits[DecimalPrefix[x, n]]]

NotPerfectSquareQ[x_] := Not[IntegerQ[Sqrt[x]]]
notperfectsquares = Select[Range[1, 100], NotPerfectSquareQ];

Total[Map[Function[x, DecimalPrefixDigitSum[Sqrt[x], 100]], notperfectsquares]]

File: codes/python/p080.py
# 
# Solution to Project Euler problem 80
# Copyright (c) Sankha Saha. All rights reserved.
# 
#  
#   
# 

import math


def compute():
	DIGITS = 100
	MULTIPLIER = 100**DIGITS
	ans = sum(
		sum(int(c) for c in str(math.isqrt(i * MULTIPLIER))[ : DIGITS])
		for i in range(100)
		if math.isqrt(i)**2 != i)
	return str(ans)


if __name__ == "__main__":
	print(compute())
